### PR TITLE
Fix for reset offset to earliest strategy

### LINKED ISSUE #8: API Error/Fetch Loop
When trying to subscribe to the remote server I mentioned in a another bug, after fixing the advertised hostname and port, I'm getting the follow loop.

```
[Kafka Logger]: Consumer: Running worker on host server1.mydomain.com:9092
[Kafka Logger]: Consumer: received API error 1 for topic testTopic:0
[Kafka Logger]: Offsets were reset. Forcing re-fetch.
[Kafka Logger]: Consumer: received API error 1 for topic testTopic:0
[Kafka Logger]: Offsets were reset. Forcing re-fetch.
[Kafka Logger]: Consumer: received API error 1 for topic testTopic:0
[Kafka Logger]: Offsets were reset. Forcing re-fetch.
[Kafka Logger]: Consumer: received API error 1 for topic testTopic:0
[Kafka Logger]: Offsets were reset. Forcing re-fetch.
[Kafka Logger]: Consumer: received API error 1 for topic testTopic:0
[Kafka Logger]: Offsets were reset. Forcing re-fetch.
[Kafka Logger]: Consumer: received API error 1 for topic testTopic:0
```

### PR DESCRIPTION
**Pull request by pulyaevskiy**
Fixes #8 .
Still need to spend some time properly exposing reset offset logic in consumer.

### CODE REVIEW

**Comment by pulyaevskiy** [comment]
cc @trickleup

File: lib/src/consumer_group.dart [hunk 1]
```diff
@@ -90,8 +90,12 @@ class ConsumerGroup {
     var earliestOffsets = await offsetMaster.fetchEarliest(topicPartitions);
     var offsets = new List<ConsumerOffset>();
     for (var earliest in earliestOffsets) {
+      // When consuming we always pass `currentOffset + 1` to fetch next
+      // message so here we need to substract 1 from earliest offset, otherwise
+      // we'll end up in an infinite loop of "InvalidOffset" errors.
+      var actualOffset = earliest.offset - 1;
       offsets.add(new ConsumerOffset(earliest.topicName, earliest.partitionId,
-          earliest.offset, 'resetToEarliest'));
+          actualOffset, 'resetToEarliest'));
```
Comment (pulyaevskiy) [inline lib/src/consumer_group.dart:98]: @trickleup this is actual fix.

**Comment by pulyaevskiy** [comment]
Example output:

```
➜  dart-kafka git:(fix-offset-reset-to-earlier) ✗ dart bin/consume.dart
[INFO] Kafka: Consumer: Running worker on host 192.168.99.100:9093
[INFO] Kafka: Consumer: Running worker on host 192.168.99.100:9092
[FINE] Kafka: Consumer: Sending fetch request.
[FINE] Kafka: Consumer: Sending fetch request.
[WARNING] Kafka: Consumer: received API error 1 for topic dartKafkaTest:1
[INFO] Kafka: Topics to reset: {dartKafkaTest: {1}}
[WARNING] Kafka: Consumer: received API error 1 for topic dartKafkaTest:0
[INFO] Kafka: Topics to reset: {dartKafkaTest: {0}}
[WARNING] Kafka: Consumer: received API error 1 for topic dartKafkaTest:2
[INFO] Kafka: Topics to reset: {dartKafkaTest: {0, 2}}
[WARNING] Kafka: Offsets were reset to OffsetOutOfRangeBehavior.resetToEarliest. Forcing re-fetch.
[WARNING] Kafka: Offsets were reset to OffsetOutOfRangeBehavior.resetToEarliest. Forcing re-fetch.
[FINE] Kafka: Consumer: Sending fetch request.
[FINE] Kafka: Consumer: Sending fetch request.
O: 9, P: 1, V: msg1
O: 17, P: 0, V: msg0
O: 9, P: 2, V: msg2
[FINE] Kafka: Consumer: Sending fetch request.
[FINE] Kafka: Consumer: Sending fetch request.
...
```

File: lib/src/consumer_group.dart (hunk 1 above)
Comment (trickleup) [inline lib/src/consumer_group.dart:98]: Looking good!

**Pull request merged**